pkg/service/one2onerestore/worker_validate_integration_test.go

```diff
@@ -0,0 +1,341 @@
+				hrt.SetInterceptor(nil)
+			},
+			expecterErr: true,
+		},
+		{
+			name: "Less nodes in source clusters",
+			manifestsProvider: func() []*backupspec.ManifestInfo {
+				return manifests[1:]
+			},
+			hostsProvider: func() []Host {
+				return hosts
+			},
+			nodeMappingsProvider: func() []nodeMapping {
+				return nodeMappings
+			},
+			expecterErr: true,
+		},
+		{
+			name: "Wrong source nodes mapping",
+			manifestsProvider: func() []*backupspec.ManifestInfo {
+				return manifests
+			},
+			hostsProvider: func() []Host {
+				return hosts
+			},
+			nodeMappingsProvider: func() []nodeMapping {
+				modified := make([]nodeMapping, len(nodeMappings))
+				copy(modified, nodeMappings)
+				modified[0].Source.DC = "not found"
+				return modified
+			},
+			setIntereptor: func() {
+				hrt.SetInterceptor(nil)
+			},
+			expecterErr: true,
+		},
+		{
+			name: "Wrong target nodes mapping",
+			manifestsProvider: func() []*backupspec.ManifestInfo {
+				return manifests
+			},
+			hostsProvider: func() []Host {
+				return hosts
+			},
+			nodeMappingsProvider: func() []nodeMapping {
+				modified := make([]nodeMapping, len(nodeMappings))
+				copy(modified, nodeMappings)
+				modified[0].Target.DC = "not found"
+				return modified
+			},
+			setIntereptor: func() {
+				hrt.SetInterceptor(nil)
+			},
+			expecterErr: true,
+		},
+		{
+			name: "Node doesn't have access to manifest location",
+			manifestsProvider: func() []*backupspec.ManifestInfo {
+				return manifests
+			},
+			hostsProvider: func() []Host {
+				return hosts
+			},
+			nodeMappingsProvider: func() []nodeMapping {
+				return nodeMappings
+			},
+			setIntereptor: func() {
+				randomNode := hosts[rand.IntN(len(hosts))].Addr
+				hrt.SetInterceptor(httpx.RoundTripperFunc(func(req *http.Request) (*http.Response, error) {
+					if req.URL.Hostname() != randomNode {
+						return nil, nil
+					}
+					if req.URL.Path != "/agent/rclone/operations/cat" {
+						return nil, nil
+					}
+					return httpx.MakeResponse(req, http.StatusBadRequest), nil
+				}))
+			},
+			expecterErr: true,
+		},
+		{
+			name: "Node is not alive",
+			manifestsProvider: func() []*backupspec.ManifestInfo {
+				return manifests
+			},
+			hostsProvider: func() []Host {
+				return hosts
+			},
+			nodeMappingsProvider: func() []nodeMapping {
+				return nodeMappings
+			},
+			setIntereptor: func() {
+				randomNode := hosts[rand.IntN(len(hosts))].Addr
+				hrt.SetInterceptor(httpx.RoundTripperFunc(func(req *http.Request) (*http.Response, error) {
+					if req.URL.Hostname() != randomNode {
+						return nil, nil
+					}
+					if req.URL.Path != "/storage_service/scylla_release_version" {
+						return nil, nil
+					}
+					return httpx.MakeResponse(req, http.StatusBadRequest), nil
+				}))
+			},
+			expecterErr: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.setIntereptor != nil {
+				tc.setIntereptor()
+			}
```

Comment:
We should always set interceptor to nil if it's not specified in the test case.
Otherwise, test case ordering could influence test results.
It would also allow for not setting nil interceptors directly in the test cases.